docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_geo_restriction_remediation.mdx

```diff
@@ -1,6 +1,321 @@
+       else:
+           print(f"Distribution {distribution_id} already has Geo Restriction enabled")
+   ```
+
+These steps and scripts will help you ensure that Geo Restriction is enabled for your CloudFront distributions using Python and Boto3.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the CloudFront console at https://console.aws.amazon.com/cloudfront/.
+
+2. In the CloudFront console, in the top pane, click on the ID of the distribution that you want to check.
+
+3. In the distribution settings page, click on the "Restrictions" tab.
+
+4. Under the Geo Restriction section, check if the "Enable Geo Restriction" option is enabled or not. If it's not enabled, then it's a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all CloudFront distributions: Use the following AWS CLI command to list all the CloudFront distributions in your AWS account:
+
+   ```
+   aws cloudfront list-distributions --query 'DistributionList.Items[*].Id' --output text
+   ```
+   This command will return a list of IDs for all the CloudFront distributions.
+
+3. Get the details of each CloudFront distribution: For each ID returned in the previous step, run the following command to get the details of the distribution:
+
+   ```
+   aws cloudfront get-distribution --id <distribution-id>
+   ```
+   Replace `<distribution-id>` with the ID of the distribution you want to check. This command will return a JSON object with the details of the distribution.
+
+4. Check for Geo Restriction: In the JSON object returned in the previous step, look for the `Restrictions` field. If the `GeoRestriction` field under `Restrictions` is not present or if the `RestrictionType` is not set to `whitelist` or `blacklist`, then the CloudFront distribution does not have Geo Restriction enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install and configure Boto3 to interact with AWS services. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials to allow Boto3 to interact with AWS services on your behalf. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID='your_access_key'
+   AWS_SECRET_ACCESS_KEY='your_secret_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>